cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -2175,6 +2175,244 @@ TEST(TestFixedSizeBinaryMinMaxKernel, Basics) {
+                 options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": null, "max": null})")));
+      EXPECT_THAT(
+          MinMax(DictArrayFromJSON(dict_ty, R"([3, 1, 1, 4, 0, 2])",
+                                   R"(["5.10", "-1.23", "2.00", "3.45", "4.56"])"),
+                 options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": "-1.23", "max": "5.10"})")));
+
+      options = ScalarAggregateOptions(/*skip_nulls=*/true, /*min_count=*/0);
+      EXPECT_THAT(MinMax(chunked, options),
+                  ResultWith(ScalarFromJSON(ty, R"({"min": "-1.23", "max": "5.10"})")));
+      EXPECT_THAT(
+          MinMax(DictArrayFromJSON(dict_ty, R"([null, null, null])", R"([])"), options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": null, "max": null})")));
+      EXPECT_THAT(MinMax(DictArrayFromJSON(dict_ty, R"([0])", R"(["1.00"])"), options),
+                  ResultWith(ScalarFromJSON(ty, R"({"min": "1.00", "max": "1.00"})")));
+
+      options = ScalarAggregateOptions(/*skip_nulls=*/true, /*min_count=*/5);
+      EXPECT_THAT(MinMax(chunked, options),
+                  ResultWith(ScalarFromJSON(ty, R"({"min": null, "max": null})")));
+      EXPECT_THAT(
+          MinMax(DictArrayFromJSON(dict_ty, R"([0, 1, 1, 0])", R"(["5.10", "-1.23"])"),
+                 options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": null, "max": null})")));
+      EXPECT_THAT(
+          MinMax(DictArrayFromJSON(dict_ty, R"([null, 3, 1, 1, 4, 0, 2, null, null])",
+                                   R"(["5.10", "-1.23", "2.00", "3.45", "4.56"])"),
+                 options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": "-1.23", "max": "5.10"})")));
+
+      options = ScalarAggregateOptions(/*skip_nulls=*/true, /*min_count=*/1);
+      EXPECT_THAT(MinMax(chunked, options),
+                  ResultWith(ScalarFromJSON(ty, R"({"min": "-1.23", "max": "5.10"})")));
+      EXPECT_THAT(
+          MinMax(DictArrayFromJSON(dict_ty, R"([null, null, null])", R"([])"), options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": null, "max": null})")));
+      EXPECT_THAT(MinMax(DictArrayFromJSON(dict_ty, R"([0])", R"(["1.00"])"), options),
+                  ResultWith(ScalarFromJSON(ty, R"({"min": "1.00", "max": "1.00"})")));
+
+      // compact dictionary
+      EXPECT_THAT(
+          MinMax(
+              DictArrayFromJSON(
+                  dict_ty, R"([3, 1, 1, 4, 0, 2])",
+                  R"(["5.10", "-1.23", "2.00", "3.45", "4.56", "8.20", "9.20", "10.20"])"),
+              options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": "-1.23", "max": "5.10"})")));
+      EXPECT_THAT(
+          MinMax(
+              DictArrayFromJSON(
+                  dict_ty, R"([5, 1, 1, 6, 0, 2])",
+                  R"(["5.10", "-1.23", "2.00", "3.45", "4.56", "8.20", "9.20", "10.20"])"),
+              options),
+          ResultWith(ScalarFromJSON(ty, R"({"min": "-1.23", "max": "9.20"})")));
+    }
```

Comment:
It would be nice to streamline these tests but also make them more comprehensive. I would expect:
- tests with nulls in indices
- tests with nulls in values
- tests with nulls in both indices and values
- tests with unreferenced values and multiply referenced values
- variations on `min_count` and `skip_nulls` options

Don't hesitate to add comments to clearly describe which is which.

Reply:
Ok, I will add comments and more test cases when code changes are completed